docs/docs/reference/other-new-features/tupled-function.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: doc-page
+title: "Tupled Function"
+---
+
+Tupled Function
+----------------------
+
+With functions bounded to arities up to 22 it was possible to generalize some operation on all function types using overloading. 
+Now that we have functions and tuples generalized to [arities above 22](https://dotty.epfl.ch/docs/reference/dropped-features/limit22.html) overloading is not an option anymore. 
+The type class `TupleFunction` provides a way to abstract directly over functions of any arity converting it to an equivalent function that receives all arguments in a single tuple.
+
+```scala
+/** Type class relating a `FunctionN[..., R]` with an equvalent tupled function `Function1[TupleN[...], R]`
+ *
+ *  @tparam F a function type
+ *  @tparam G a tupled function type (function of arity 1 receiving a tuple as argument)
+ */
+@implicitNotFound("${F} cannot be tupled as ${G}")
+sealed trait TupledFunction[F, G] {
+  def tuple(f: F): G
+}
+```
+
+The compiler will synthesize an instance of `TupledFunction[F, G]` if:
```

Comment:
Should we also indicate that if the type `G` is not instantiated, the compiler will instantiate it so that it is aligned with the type of `F`?

Reply:
(actually, it seems to work in both directions: if one of `F` or `G` is unknown, the compiler computes it according to the other type)

Reply:
Yes, it works in both directions. I had to make it infer `F` from `G` to make `untulpe` work

Reply:
I added it in the documentation